src/test/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotTest.java

```diff
@@ -77,6 +77,32 @@ public void testSnapshotRecordConsumption(boolean colocation) throws Exception {
           }).get();
     }
 
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    public void testSnapshotRecordCountInInitialOnlyMode(boolean colocation) throws Exception {
+        setCommitCallbackDelay(10000);
+        createTables(colocation);
+        final int recordsCount = 4000;
+        insertBulkRecords(recordsCount, "public.test_1");
+
+        LOGGER.info("Creating DB stream ID");
+        String dbStreamId = TestHelper.getNewDbStreamId(DEFAULT_COLOCATED_DB_NAME, "test_1");
+        Configuration.Builder configBuilder =
+          TestHelper.getConfigBuilder(DEFAULT_COLOCATED_DB_NAME, "public.test_1", dbStreamId);
+        configBuilder.with(YugabyteDBConnectorConfig.SNAPSHOT_MODE, YugabyteDBConnectorConfig.SnapshotMode.INITIAL_ONLY.getValue());
+        startEngine(configBuilder);
+
+        awaitUntilConnectorIsReady();
+
+        // Only verifying the record count since the snapshot records are not ordered, so it may be
+        // a little complex to verify them in the sorted order at the moment
+        CompletableFuture.runAsync(() -> verifyRecordCount(recordsCount))
```

Comment:
I was thinking of removing the instances of `CompletableFuture` altogether. I mean, it's not relevant now in the tests to perhaps it would be a good idea not to use this in new tests from now on.

Simply using `verifyRecordCount(recordsCount);` should do the job.